### DISCUSSION TITLE
Advice on how best to re-authenticate running docker containers

### DISCUSSION THREAD

**Discussion by CEardley** [icloud-photos-downloader/icloud_photos_downloader#800] [Q&A] [answered]
So I've been playing around with this tool over the weekend and have an instance of icloudpd running in a docker container on a local server (Raspberry Pi connected to RAID array). The plan for my final setup is to have two docker containers running icloudpd for two different iCloud accounts, checking once per day, with a third container running an SMTP server to act as an email host for re-authentication notifications. 

The main question I have is how best to re-authenticate the running docker containers? My first thought is to docker exec into the running containers and run `icloudpd --username my@email.address --password my_password --auth-only`, but to do that I think I'll need to make a custom docker image with bash installed. Wonder if there's a better approach to re-authenticating?

**Comment by AndreyNikiforov** [comment, marked as answer]
> The main question I have is how best to re-authenticate the running docker containers? ... Wonder if there's a better approach to re-authenticating?

You can run icloudpd with `--watch-with-interval` -- you would not need to re-authenticate for days/weeks/months. Note that setting interval too high may trigger Apple resetting auth cookies and, thus, require re-authentication - I never experimented with "max interval" and 3600 works fine for me.

**Reply by CEardley** [reply]
@AndreyNikiforov Thanks for the response. I assume that you're also suggesting that rather than re-authenticate a running container in a couple of weeks, I simply restart the container? (Honestly don't know why I hadn't thought of that but that makes so much sense!)

**Comment by AndreyNikiforov** [comment]
> ... rather than re-authenticate a running container in a couple of weeks, I simply restart the container? 

No exactly "restart the container", you'll get container running (with `--watch-with-interval` parameter) non-stop for days/weeks/months and it will keep your collection synchronized from iCloud to your local storage without requiring re-authentication (eventually Apple expires cookies and you will have to start docker again and authenticate).

**Reply by CEardley** [reply]
Wonderful, thanks @AndreyNikiforov
